src/main/java/uk/ac/cam/cl/dtg/isaac/api/IsaacController.java

```diff
@@ -205,6 +205,11 @@ public final Response search(@Context final Request request, @Context final Http
                 showHiddenContent = isUserStaff(userManager, (RegisteredUserDTO) currentUser);
             }
             List<String> documentTypes = !types.isEmpty() ? Arrays.asList(types.split(",")) : null;
+            // Return an error if any of the proposed document types are invalid
+            if (documentTypes != null && !SITE_WIDE_SEARCH_VALID_DOC_TYPES.containsAll(documentTypes)) {
```

Comment:
Does this work if `documentTypes` is empty? I assume so?

Reply:
Great! I think the comma only case will cause two empty strings, and they aren't in the set so it is all okay 🎉